pkg/services/datasources/models.go

```diff
@@ -134,6 +134,7 @@ type UpdateDataSourceCommand struct {
 	ReadOnly                bool              `json:"-"`
 	EncryptedSecureJsonData map[string][]byte `json:"-"`
 	UpdateSecretFn          UpdateSecretFn    `json:"-"`
+	SecureJsonOverwrite     bool              `json:"-"`
```

Comment:
Suggestion: `IgnoreOldSecureJsonData`? If I set `SecureJsonOverwrite: false` it could mean that a new password (for example) wouldn't overwrite the current one, not that it deletes unset params.

Feel free to keep the naming you think is best though :+1: